webpack.test.js

```diff
@@ -0,0 +1,145 @@
+
+const BASE_HREF = process.env.BASE_HREF || '/';
+
+module.exports = {
+  mode: 'development',
+  entry: './src/index.test.js',
+  devtool: "source-map",
+  output: {
+    filename: 'bundle-test.js',
+    path: path.resolve(__dirname, 'dist'),
+  },
+  resolve: {
+    alias: {
+      'moment/moment.js': momentPath,
+      'moment$': momentPath
+    },
+  },
+  plugins: [
+    new webpack.IgnorePlugin({ resourceRegExp: /codemirror/ }), // for summernote
+    new webpack.NormalModuleReplacementPlugin(
+      /node_modules\/esn-frontend-common-libs\/src\/frontend\/js\/modules\/i18n\/i18n-loader.service.js/,
+      i18nLoaderMockPath
+    ),
+    new webpack.ProvidePlugin({
+      jQuery: 'jquery',
+      'window.jQuery': 'jquery',
+      'Chart': chartJs,
+      chai: chaiPath,
+      materialAdmin: materialAdmin,
+      angular: angularCommon,
+      _: lodashPath,
+     moment: momentPath,
+     ICAL: icalPath,
+      'window.angularInjections': angularInjections,
+      localforage: 'localforage', // for calendar
+    }),
+    /*
+     * To transform assets/index.pug to an HTML file, with webpack autoimporting the "main.js" bundle
+     */
+    new HtmlWebpackPlugin({
+      template: './assets/index.pug',
+      filename: './index.html'
+    }),
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.(woff(2)?|ttf|eot|svg)(\?v=\d+\.\d+\.\d+)?$/,
+        use: [
+          {
+            loader: 'file-loader',
+          },
+        ],
+      },
+      {
+        test: /all\.less$/,
+        use: [
+          {
+            loader: 'style-loader', // creates style nodes from JS strings
+          },
+          {
+            loader: 'css-loader', // translates CSS into CommonJS
+          },
+          {
+            loader: 'less-loader', // compiles Less to CSS
+            options: {
+              lessOptions: {
+                javascriptEnabled: true
+              }
+            }
+          }
+        ]
+      },
+      {
+        test: /\.(png|jpe?g|gif)$/i,
+        use: [
+          {
+            loader: 'url-loader',
+          },
+        ],
+      },
+      {
+        test: /\.svg$/,
+        loader: 'svg-inline-loader'
+      },
+      /*
+      * for the "index.html" file of this SPA.
+      *
+      */
+      {
+        test: /assets\/index\.pug$/,
+        use: [
+          {
+            loader: 'html-loader',
+          },
+          {
+            loader: 'pug-html-loader',
+            options: {
+              data: {
+                base: BASE_HREF,
+              },
+            },
+          },
+        ],
+      },
+      {
+        test: /\.pug$/i,
+        exclude: /assets\/index\.pug$/,
+        use: [
+          {
+            loader: 'apply-loader',
+          },
+          {
+            loader: 'pug-loader',
+            options: pugLoaderOptions
+          },
+        ],
+      },
+      { 
+        test: /\.(js|jsx)$/,
+        use: [
+          {
+            loader: "babel-loader"
```

Comment:
Question for all: Do we really need `babel` to run tests? I assume that the chosen browser headless support ES6

@MichaelBailly @chamerling @huy-ta @piavgh @ppppbn

Reply:
Tested without `babel` and the performance improved a lot. I think we can get rid of `babel`

Reply:
No we shouldn't need babel for tests.